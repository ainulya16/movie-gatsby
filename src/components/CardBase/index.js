import React from 'react';
import { any, bool, func, arrayOf, string, number } from 'prop-types';
import styled, { css } from 'styled-components';
import { Accordion, Button } from '@aia-digital/ui-library';
import { theme } from '../../utils/constants';
import icDiamond from '../../assets/images/icDiamond.svg';
import icClassic from '../../assets/images/icClassic.svg';

const Container = styled.div`
  ${props =>
    props.isRecommended &&
    css`
      border: 3px solid ${theme.color.p1};
    `}
  font-family:'AIARegular';
  box-shadow: 0 0 10px 2px rgba(0, 0, 0, 0.2);
  border-radius: 10px 10px;
  max-width: 380px;
  margin: auto;
  padding-top: 12px;
  text-align: center;
  overflow: hidden;
  .padding-s {
    width: auto;
    display: inline-block;
    border-bottom: 1px solid #d31546;
    padding: 8px;
    margin-bottom: 7px;
  }
  @media (max-width: 1024px) {
    width: 315px;
  }
`;

const BorderedContainer = styled.div`
  border: 1px solid red;
`;
BorderedContainer.displayName = 'BorderedContainer';

const Title = styled.div`
  margin-top: 6px;
  margin-bottom: 20px;
  text-align: center;
  .iconPackage {
    display: inline-block;
    text-align: center;
    margin-bottom: 6.5px;
    img {
      width: 30px;
      height: 32.5px;
    }
  }
  .packageName {
    font-family: 'AIAMedium';
    font-size: 20px;
    color: #685556;
    line-height: 1;
    letter-spacing: normal;
  }
  .redText {
    color: #d31546;
  }
  @media(max-width: 768px) {
    margin-top: 8px;
    .iconPackage {
      img {
        width: 36px;
        height: 30px;
      }
    }
  }
`;
const TitleAmount = styled.div`
  text-align: center;
  font-family: 'AIAMedium';
  color: #685556;
  line-height: normal;
  letter-spacing: normal;
  .small1 {
    font-size: 15px;
    margin-bottom: 2px;
  }
  .small2 {
    font-size: 22px;
  }
`;
const Price = styled.div`
  font-family: 'AIAMedium';
  text-align: center;
  font-size: 15px;
  color: #685556;
  line-height: 1;
  letter-spacing: normal;
  .amount {
    font-family: AIACondensedMedium;
    font-size: 34px;
    margin-bottom: 5px;
  }
  .term {
    margin-top: -3x;
    margin-bottom: -3px;
    line-height: 1.07;
  }
`;

const AccordionWrapper = styled.div`
  text-align: left;
  margin-top: 20px;
  line-height: normal;
  letter-spacing: -0.45px;
  .faqText > div > div > div {
     position: relative;
     bottom: 2px;
     right: -5px;
  }
  .faqText > div > div > div i{
     color: #596c80;
     width: 10px;
     height: 16px;
  }
  .faqText > div > div {
    position: relative;
    left: -6px;
    margin-bottom: -8px;
    padding-right: 29px;
  }
  .faqText > div {
    padding: 10px 0px 0px 0px;
  }
  .faqText > div > div:nth-child(2) {
    padding-bottom: 15px;
  }
  .itemBenefit {
    font-family: AIARegular;
    font-size: 16px;
    line-height: 1.5;
    letter-spacing: -0.45px;
    color: #685556;
  }
`;

const ButtonWrapper = styled.div`
  width: 100%;
  text-align: center;
  margin-top: 16px;
  margin-bottom: 10px;
  button {
    width: 356px;
    height: 50px;
  }
  div:nth-child(1) {
    width: auto;
  }
  @media(max-width: 1024px){
    margin-top: 19.6px;
    button {
      width: 295px;
    }
  }
`;

const StyledButton = styled(Button)`
  margin: auto;
  padding-left: 50px;
  padding-right: 50px;
`;

const renderAccordion = benefits => {
  const dataFormat = benefits.map(item => {
    const icon = item.description ? { rotate: true, name: 'forward', size: 'm'} :  null;
    const title = (
      <div className="itemBenefit">
        <span style={{ fontWeight: 'bold' }}>
          S${new Intl.NumberFormat().format(item.value)}{' '}
        </span>
        <span>{item.name.en}</span>
      </div>
    );
    return ({ title, content: item.description ? item.description.en : null, icon});
  });
  return <Accordion className="faqText" data={dataFormat} />;
};

const Card = ({
  name,
  totalPremium,
  mode,
  benefits,
  onButtonClicked,
  showButton,
  titleButton,
  maximumInsured,
  ...attr
}) => {
  return (
    <Container {...attr}>
      <div className="padding-s">
        <Title>
          <div className="iconPackage">
            <img src={name === 'Diamond' ? icDiamond : icClassic} alt="" />
          </div>
          <div
            className={
              name === 'Diamond' ? 'packageName redText' : 'packageName'
            }
          >
            {name}
          </div>
        </Title>
        <Price>
          <div className="amount">S$ {new Intl.NumberFormat('en-SG', {minimumFractionDigits: 2}).format(totalPremium)}</div>
          <div className="term">{`${mode.toLowerCase()}`}</div>
        </Price>
      </div>
      <TitleAmount>
        <div className="small1">Maximum Insured amount</div>
        <div className="small2">
          S$ {new Intl.NumberFormat().format(maximumInsured)}
        </div>
      </TitleAmount>
      {benefits.length > 0 && (
        <AccordionWrapper>{renderAccordion(benefits)}</AccordionWrapper>
      )}
      {showButton && (
        <ButtonWrapper>
          <StyledButton gradient rounded onClick={onButtonClicked}>{titleButton}</StyledButton>
        </ButtonWrapper>
      )}
    </Container>
  );
};

Card.propTypes = {
  showButton: bool,
  onButtonClicked: func,
  name: string.isRequired,
  totalPremium: number.isRequired,
  maximumInsured: number.isRequired,
  mode: string.isRequired,
  benefits: arrayOf(any).isRequired,
  isRecommended: bool,
  titleButton: string,
};

Card.defaultProps = {
  showButton: true,
  onButtonClicked: () => {},
  isRecommended: false,
  titleButton: 'SELECT',
};

export default Card;
