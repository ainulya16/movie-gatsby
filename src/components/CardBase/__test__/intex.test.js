import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Accordion } from '@aia-digital/ui-library';
import CardBase from '../index';

let renderComponent;
const cardProps = {
  showButton: true,
  onButtonClicked: jest.fn(),
  name: 'Classic',
  totalPremium: 8.1,
  mode: 'monthly',
  benefits: [{name:{en:'name1'}, description:{en:'description1'}, value: 2000}],
  isRecommended: false,
  titleButton: 'SELECT',
};
describe('<CardBase />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<CardBase {...cardProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  beforeAll(() => {
    renderComponent = mount(<CardBase {...cardProps} />);
  });
  it('should have showButton prop', () => {
    expect(renderComponent.prop('showButton')).toBe(cardProps.showButton);
  });
  it('should have onButtonClicked prop', () => {
    expect(renderComponent.prop('onButtonClicked')).toEqual(cardProps.onButtonClicked);
  });
  it('should have name prop', () => {
    expect(renderComponent.prop('name')).toEqual(cardProps.name);
  });
  it('should have totalPremium prop', () => {
    expect(renderComponent.prop('totalPremium')).toEqual(cardProps.totalPremium);
  });
  it('should have mode prop', () => {
    expect(renderComponent.prop('mode')).toEqual(cardProps.mode);
  });
  it('should have benefits prop', () => {
    expect(renderComponent.prop('benefits')).toEqual(cardProps.benefits);
  });
  it('should have isRecommended prop', () => {
    expect(renderComponent.prop('isRecommended')).toEqual(cardProps.isRecommended);
  });
  it('should have titleButton prop', () => {
    expect(renderComponent.prop('titleButton')).toEqual(cardProps.titleButton);
  });
  it('should not display Accordion when benefit data is empty', () => {
    renderComponent = mount(<CardBase mode="test" benefits={[]} />);
    expect(renderComponent.find(Accordion)).toEqual({});
  });
});
