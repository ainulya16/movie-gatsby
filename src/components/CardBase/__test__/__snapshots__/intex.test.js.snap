// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBase /> should match snapshot 1`] = `
.c10 {
  color: #000;
  font-size: 2em;
}

.c5 {
  width: 100%;
}

.c6 {
  border-bottom: solid 1px #dbb680;
  padding: 15px 0px;
}

.c12 {
  display: none;
  -webkit-transition: display 0.5s ease-in-out;
  transition: display 0.5s ease-in-out;
  margin-left: 50px;
}

.c8 {
  width: 50px;
  min-width: 50px;
  text-align: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c9 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 p {
  font-size: 18px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  min-width: 250px;
  width: 250px;
  text-align: center;
}

.c16 {
  height: 45px;
  font-family: 'AIAMedium';
  text-transform: none;
  font-size: 14px;
  border: 0px;
  display: inline-block;
  background: #D31145;
  -webkit-transition: width 0.3s ease,min-width 0.3s ease,background-color 0.3s ease,border-radius 0.3s ease,border 0.3s ease,display 0.3s ease;
  transition: width 0.3s ease,min-width 0.3s ease,background-color 0.3s ease,border-radius 0.3s ease,border 0.3s ease,display 0.3s ease;
  color: #FFFFFF;
  background-color: #FFF;
  background-image: linear-gradient(to bottom,#d31145,#D31145,#d31145);
  border: none;
  border-radius: 5px;
}

.c16:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
  -webkit-transition: scale 0.1s ease-in-out;
  transition: scale 0.1s ease-in-out;
}

.c16:focus {
  outline: none;
}

.c16:disabled {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 0.4;
  -webkit-filter: alpha(opacity=40);
  filter: alpha(opacity=40);
}

.c0 {
  font-family: 'AIARegular';
  box-shadow: 0 0 10px 2px rgba(0,0,0,0.2);
  border-radius: 10px 10px;
  max-width: 380px;
  margin: auto;
  padding-top: 12px;
  text-align: center;
  overflow: hidden;
}

.c0 .padding-s {
  width: auto;
  display: inline-block;
  border-bottom: 1px solid #d31546;
  padding: 8px;
  margin-bottom: 7px;
}

.c1 {
  text-align: center;
}

.c1 .packageName {
  font-family: 'AIAMedium';
  font-size: 20px;
  color: #685556;
  margin-bottom: 20px;
}

.c1 .iconPackage {
  display: inline-block;
  text-align: center;
  margin-bottom: -3.5px;
}

.c1 .iconPackage img {
  width: 36px;
  height: 30px;
}

.c1 .redText {
  color: #d31546;
}

.c3 {
  text-align: center;
  font-family: 'AIAMedium';
  color: #685556;
}

.c3 .small1 {
  font-size: 15px;
}

.c3 .small2 {
  font-size: 22px;
}

.c2 {
  font-family: 'AIAMedium';
  text-align: center;
  font-size: 15px;
  margin-top: -10px;
  color: #685556;
}

.c2 b {
  font-family: AIACondensed;
  font-size: 32px;
}

.c2 .term {
  margin-top: -3x;
  margin-bottom: -3px;
}

.c4 {
  text-align: left;
  margin-top: 15px;
}

.c4 .faqText > div > div > div {
  position: relative;
  bottom: 4px;
  right: -5px;
}

.c4 .faqText > div > div > div i {
  color: #596c80;
  width: 10px;
  height: 16px;
}

.c4 .faqText > div > div {
  position: relative;
  left: -6px;
  margin-bottom: -8px;
  padding-right: 29px;
}

.c4 .faqText > div {
  padding: 10px 0px 0px 0px;
}

.c4 .faqText > div > div:nth-child(2) {
  padding-bottom: 15px;
}

.c4 .itemBenefit {
  font-family: AIARegular;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: -0.45px;
  -moz-letter-spacing: -0.45px;
  -ms-letter-spacing: -0.45px;
  letter-spacing: -0.45px;
  color: #685556;
}

.c13 {
  width: 100%;
  text-align: center;
  margin-top: 16px;
  margin-bottom: 10px;
}

.c13 button {
  width: 356px;
}

.c13 div:nth-child(1) {
  width: unset;
}

.c15 {
  margin: auto;
  padding-left: 50px;
  padding-right: 50px;
}

.c11 {
  -webkit-letter-spacing: 0.23px;
  -moz-letter-spacing: 0.23px;
  -ms-letter-spacing: 0.23px;
  letter-spacing: 0.23px;
  font-family: AIAMedium;
  color: #333333;
  text-transform: unset;
}

@media (max-width:1024px) {
  .c0 {
    width: 315px;
  }
}

@media (max-width:1024px) {
  .c13 button {
    width: 295px;
  }
}

<div
  className="c0"
>
  <div
    className="padding-s"
  >
    <div
      className="c1"
    >
      <div
        className="iconPackage"
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div
        className="packageName"
      >
        Classic
      </div>
    </div>
    <div
      className="c2"
    >
      <b>
        S$ 
        8.1
      </b>
      <div
        className="term"
      >
        monthly
      </div>
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="small1"
    >
      Maximum Insured amount
    </div>
    <div
      className="small2"
    >
      S$ 
      NaN
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="faqText c5"
      style={null}
    >
      <div
        className="c6"
      >
        <div
          className="c7"
          onClick={[Function]}
        >
          <div
            className="c8"
          >
            <div
              className="c9"
              rotate={false}
            >
              <i
                className="icon-forward  c10"
                color="#000"
                size="m"
              />
            </div>
          </div>
          <p
            className="c11"
            fontFamily="AIAMedium"
          >
            <div
              className="itemBenefit"
            >
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                S$
                2,000
                 
              </span>
              <span>
                name1
              </span>
            </div>
          </p>
        </div>
        <div
          className="c12"
        >
          description1
        </div>
      </div>
    </div>
  </div>
  <div
    className="c13"
  >
    <div
      className="c14"
      style={Object {}}
    >
      <button
        className="btn c15 c16"
        color="#FFFFFF"
        disabled={false}
        loading={false}
        onClick={[MockFunction]}
        radius={null}
      >
        <div
          className="child"
        >
          SELECT
        </div>
      </button>
    </div>
  </div>
</div>
`;
